featuretools/tests/entityset_tests/test_ww_es.py

```diff
@@ -715,12 +735,32 @@ def test_update_dataframe_dont_recalculate_last_time_index(es):
 
     new_dataframe = es['customers'].copy()
     new_dataframe['signup_date'] = new_time_index
+    new_dataframe.pop('_ft_last_time')
 
     es.update_dataframe('customers', new_dataframe, recalculate_last_time_indexes=False)
-    pd.testing.assert_series_equal(to_pandas(es['customers']['last_time'], sort_index=True), to_pandas(original_last_time_index, sort_index=True))
+    assert 'last_time_index' not in es['customers'].ww.metadata
+    assert original_lti_name not in es['customers'].columns
+
+
+def test_update_dataframe_recalculate_last_time_index_not_present(es):
+    es.add_last_time_indexes()
+
+    original_time_index = es['log']['datetime'].copy()
+
+    if ks and isinstance(original_time_index, ks.Series):
+        new_time_index = ks.from_pandas(original_time_index.to_pandas() + pd.Timedelta(days=10))
+    else:
+        new_time_index = original_time_index + pd.Timedelta(days=10)
+
+    new_dataframe = es['log'].copy()
+    new_dataframe['datetime'] = new_time_index
+    new_dataframe.pop('_ft_last_time')
+
+    es.update_dataframe('log', new_dataframe, recalculate_last_time_indexes=True)
+    pd.testing.assert_series_equal(to_pandas(es['log']['datetime']).sort_index(), to_pandas(new_time_index).sort_index(), check_names=False)
 
 
-def test_update_dataframe_recalculate_last_time_index(es):
+def test_update_dataframe_recalcuate_last_time_index_present(es):
```

Comment:
`recalcuate` -> `recalculate`

Reply:
Also, if I'm following this test correctly, the original last time index for the log entity is the same as the recalculated last time index. I'm thinking we might want to do something here that causes the last time index after `update_dataframe` to change from what it was before - otherwise I'm not sure we can actually verify that it was recalculated as opposed to just carried over from before.

Reply:
Yep, good call. Added a line where we confirm that the last time index column also matches the `new_time_index` here and in the test above this (which is the same except for whether or not the lti column is on the new dataframe).